plugins/InspectionGadgets/InspectionGadgetsAnalysis/src/com/siyeh/ig/controlflow/IfStatementWithIdenticalBranchesInspection.java

```diff
@@ -641,21 +642,22 @@ private static ImplicitElseData getIfWithImplicitElse(@NotNull PsiIfStatement if
       if (enclosingIf == null) break;
       currentIf = enclosingIf;
     }
-    if(conditionHasSideEffects && ifStatement != currentIf) return null;
+    if (conditionHasSideEffects && ifStatement != currentIf) return null;
     // ensure it is last statements in method
     PsiElement parent = currentIf.getParent();
     if (!(parent instanceof PsiCodeBlock)) return null;
-    if (!(parent.getParent() instanceof PsiMethod)) return null;
-    if (!statements.isEmpty()) {
-      if (PsiTreeUtil.getNextSiblingOfType(statements.get(statements.size() - 1), PsiStatement.class) != null) return null;
+    if (PsiKeyword.RETURN.equals(jumpKeyword)) {
+      if (!(parent.getParent() instanceof PsiMethod)) return null;
```

Comment:
(just passing by)
By the way probably it would be good to support lambdas as well? It looks like they aren't supported now.

Reply:
Yes, I planned to add this in another PR with other related features (handles several levels of indentation). Tell me if you want it in this PR.

Reply:
Btw supporting `yield` would also be nice

Reply:
`yield` can't be implicit

Reply:
Yes, but can be explicit:
```java
if (foo) yield 1;
yield 1;
```
Will we have a warning in this case?